appserver/monitoring-console/core/src/main/java/fish/payara/monitoring/runtime/MonitoringConsoleRuntimeImpl.java

```diff
@@ -0,0 +1,336 @@
+ * console with the Payara server.
+ * 
+ * The most complicated aspect about the implementation is the way it is bootstrapped. By implementing
+ * {@link ApplicationLifecycleInterceptor} it forces the creation of an instance of this {@link Service} even though it
+ * is not otherwise referenced within the HK2 context. As this happens fairly early in the bootstrapping it then
+ * registers itself as an {@link EventListener} so that it can run its actual {@link #init()} bootstrapping as soon as
+ * the {@link EventTypes#SERVER_READY} is received. This makes sure the bootstrapping of the console runtime does not
+ * alter the order of services created by starting to collect data from services that implement
+ * {@link MonitoringDataSource} or {@link MonitoringWatchSource}.
+ * 
+ * @author Jan Bernitt
+ * @since 5.201
+ */
+@Service
+public class MonitoringConsoleRuntimeImpl
+        implements ConfigListener, ApplicationLifecycleInterceptor, EventListener,
+        MonitoringConsoleRuntime, MonitoringConsoleWatchConfig, GroupDataRepository {
+
+    private static final Logger LOGGER = Logger.getLogger("monitoring-console-core");
+
+    private static final String SET_MONITORING_CONSOLE_CONFIGURATION_COMMAND = "set-monitoring-console-configuration";
+
+    /**
+     * The topic name used to share data of instances with the DAS.
+     */
+    private static final String MONITORING_DATA_TOPIC_NAME = "payara-monitoring-data";
+
+    @Inject
+    private PayaraExecutorService executor;
+    @Inject
+    private ServerEnvironment serverEnv;
+    @Inject @Named(ServerEnvironment.DEFAULT_INSTANCE_NAME)
+    private Config serverConfig;
+    @Inject
+    private Domain domain;
+    @Inject
+    private CommandRunner commandRunner;
+    @Inject
+    private InternalSystemAdministrator kernelIdentity;
+    @Inject
+    private HazelcastCore hazelcastCore;
+    @Inject
+    private RequestTracingService requestTracingService;
+    @Inject
+    private ServiceLocator serviceLocator;
+    @Inject
+    private Events events;
+
+    private final AtomicBoolean initialised = new AtomicBoolean();
+    private ITopic<byte[]> exchange;
+    private MonitoringConsoleConfiguration config;
+    private MonitoringConsole console;
+
+    @PostConstruct
+    public void postConstruct() {
+        events.register(this);
+    }
+
+    @Override
+    public void event(Event<?> event) {
+        if (event.is(EventTypes.SERVER_READY)) {
+            init();
+        }
+    }
+
+    public void init() {
+        if (!initialised.compareAndSet(false, true) ) {
+            return;
+        }
+        try {
+            LOGGER.info("Bootstrapping Monitoring Console Runtime");
+            boolean isDas = serverEnv.isDas();
+            config = domain.getExtensionByType(MonitoringConsoleConfiguration.class);
+            String instanceName = "server"; // default
```

Comment:
Am I understanding this correctly? It seems off to me.

So if Hazelcast isn't enabled, despite an instance for example being called "Insty1", it will go to the `MonitoringConsoleFactory` and eventually get given an `InMemorySeriesRepository` which registers the instance name as _server_ and adds that name to its set of instances. 

Is that correct? I haven't dug deep down into the guts but an instance registering itself with an incorrect name perked an eyebrow 🤔

Reply:
So far I extracted the name from hazelcast from the `HazelcastCore.INSTANCE_ATTRIBUTE` as I found that is done at other places. Happy to use some other reliable source if you can point me at one.

Reply:
I believe `ServerEnvironment.getInstanceName()` is fairly reliable.
Ottomh I'm not sure that the name returned by that is 100% always the name used in Hazelcast, but I'm nitpicking what happens if Hazelcast isn't enabled here so should be fine (assuming it works).

Although having said all that - does the monitoring even work without Hazelcast?

Reply:
I changed it to `ServerEnvironment.getInstanceName()`. It does work without Hazelcast in a single instance (DAS) configuration.